gensim/models/doc2vec_corpusfile.pyx

```diff
@@ -469,8 +469,8 @@ def d2v_train_epoch_dm_concat(
             effective_words = 0
 
             doc_words = input_stream.read_sentence()
-            _doc_tag = total_documents
-            c.doctag_len = _doc_tag < c.docvecs_count
+            _doc_tag = total_documents  # WTF? skipping to end no matter what start_doctag passed in?
```

Comment:
`FIXME` easier to grep than `WTF`. Plus a ticket, so we don't forget to clean before release if indeed critical.

Reply:
I'm not sure if it's critical. It just looks sketchy, & at a glance it's hard to imagine it doing anything - but this mode does in fact seem to pass at least 2 tiny functional tests (`TestDoc2VecModel.test_dmc_hs_fromfile` & `TestDoc2VecModel. test_dmc_neg_fromfile`). So maybe it's fine. I just want to leave some warning to anyone looking at it in the future. (It's unlikely to be me, as I strongly think the `corpus_file` duplicated-paths should be discarded in favor of some other more maintainable approach using far less code-duplication.)